spikes/lk_config_kit/main.cpp

```diff
@@ -0,0 +1,70 @@
+// Leka - LekaOS
+// Copyright 2020 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstddef>
+#include <filesystem>
+
+#include "drivers/BufferedSerial.h"
+#include "rtos/ThisThread.h"
+
+#include "ConfigKit.h"
+#include "ConfigList.h"
+#include "FATFileSystem.h"
+#include "FileManagerKit.h"
+#include "HelloWorld.h"
+#include "LogKit.h"
+#include "SDBlockDevice.h"
+
+using namespace leka;
+using namespace std::chrono_literals;
+
+SDBlockDevice sd_blockdevice(SD_SPI_MOSI, SD_SPI_MISO, SD_SPI_SCK);
+FATFileSystem fatfs("fs");
+
+auto configkit = ConfigKit();
+
+void initializeSD()
+{
+	constexpr auto default_sd_blockdevice_frequency = uint64_t {25'000'000};
+
+	sd_blockdevice.init();
+	sd_blockdevice.frequency(default_sd_blockdevice_frequency);
+
+	fatfs.mount(&sd_blockdevice);
+}
+
+auto main() -> int
+{
+	logger::init();
+
+	log_info("Hello, World!\n\n");
+
+	auto start = rtos::Kernel::Clock::now();
+
+	auto hello = HelloWorld();
+
+	rtos::ThisThread::sleep_for(1s);
+
+	hello.start();
+
+	initializeSD();
+
+	auto i = uint8_t {1};
+
+	rtos::ThisThread::sleep_for(1s);
+
+	while (true) {
+		auto t = rtos::Kernel::Clock::now() - start;
+		log_info("A message from your board %s --> \"%s\" at %i s", MBED_CONF_APP_TARGET_NAME, hello.world,
+				 int(t.count() / 1000));
+
+		if (auto write = configkit.write(i++, ConfigList::bootloader_battery_level_hysteresis); !write) {
```

Comment:
J'aurai tendance à donner d'abord l'argument du fichier avant de donner la data plutôt que l'inverse, c'est plus intuitif d'indiquer le contenant avant le contenu

```suggestion
		if (auto write = configkit.write(ConfigList::bootloader_battery_level_hysteresis, i++); !write) {
```

Reply:
je suis d'accord! 

@MMyster il y a aussi un code smell pour `i++`

je pense que tu veux `++i` et **pas** `i++` et ensuite il faut faire l'assignment avant comme demande sonarcloud

see https://gcc.godbolt.org/z/11zzfcha4